clang/test/CXX/drs/dr25xx.cpp

```diff
@@ -81,6 +81,40 @@ using ::dr2521::operator""_div;
 #endif
 } // namespace dr2521
 
+namespace dr2518 { // dr2518: 17 review
+
+template <class T>
+void f(T t) {
+  if constexpr (sizeof(T) != sizeof(int)) {
+    static_assert(false, "must be int-sized"); // expected-error {{must be int-size}}
+  }
+}
+
+void g(char c) {
+  f(0);
+  f(c); // expected-note {{requested here}}
+}
+
+template <typename Ty>
+struct S {
+  static_assert(false); // expected-error {{static assertion failed}}
+};
+
+template <>
+struct S<int> {};
+
+template <>
+struct S<float> {};
+
+int test_specialization() {
+  S<int> s1;
+  S<float> s2;
+  S<double> s3; // expected-note {{in instantiation of template class 'dr2518::S<double>' requested here}}
+}
+
+}
+
+
```

Comment:
The same namespace exist in line 24 above, but that code includes modifications to the original test that are not part of this repeated section, like the ones introduced in 5ce5e983f82c8 which should avoid one of the failures we are experiencing in the tests.